src/v/pandaproxy/schema_registry/requests/compatibility.h

```diff
@@ -27,6 +28,8 @@ void rjson_serialize(
     w.StartObject();
     w.Key("is_compatible");
     ::json::rjson_serialize(w, res.is_compat);
+    w.Key("messages");
+    ::json::rjson_serialize(w, res.messages);
```

Comment:
Is this list supposed to always be rendered, or only on some condition such as `!messages.empty()`, or `is_compatible == false`, or `verbose == true`?

Reply:
It looks like it's only returned when `verbose == true` by the reference implementation.

(Genuine question) Do you think it matters? I'm tempted to leave this as-is to keep the code simpler because I doubt returning an extra field would be problematic.

Reply:
> (Genuine question) Do you think it matters? 

I'd err on the side of following the reference implementation.